tools/tpp-opt/tpp-opt.py

```diff
@@ -0,0 +1,408 @@
+  mod = ApplyRegisteredPass(mod, "convert-vector-to-scf")
+  mod = ApplyRegisteredPass(mod, "arith-expand")
+  mod = ApplyRegisteredPass(mod, "lower-affine")
+
+  if "late" in config["dump"]:
+    transform.PrintOp(target=mod, name="DUMP: stage-late")
+
+  # Lower to LLVM
+  # TODO: call libxsmm to figure out if AMX is available
+  options = f"amx={int(has_amx)}"
+  mod = ApplyRegisteredPass(mod, "convert-vector-to-llvm", options=options)
+  mod = ApplyRegisteredPass(mod, "finalize-memref-to-llvm")
+  mod = ApplyRegisteredPass(mod, "convert-scf-to-cf")
+  if def_parallel:
+    mod = ApplyRegisteredPass(mod, "convert-openmp-to-llvm")
+  mod = ApplyRegisteredPass(mod, "convert-math-to-llvm")
+
+  if gpu_backend:
+    func = Match(mod, ops={"func.func"})
+    ApplyRegisteredPass(func, "gpu-async-region")
+    assert False
+    # gpu-to-llvm cannot be invoked from transform-interpreter as it
+    # tries to load ... something while multi-threaded PassManager is running.
+    mod = ApplyRegisteredPass(mod, "gpu-to-llvm")
+    mod = ApplyRegisteredPass(
+      mod, "gpu-module-to-binary", options="compilation-target=fatbin"
+    )
+    mod = ApplyRegisteredPass(mod, "async-to-async-runtime")
+    mod = ApplyRegisteredPass(mod, "async-runtime-ref-counting")
+    mod = ApplyRegisteredPass(mod, "convert-async-to-llvm")
+
+  mod = ApplyRegisteredPass(mod, "convert-math-to-llvm")
+  mod = ApplyRegisteredPass(mod, "convert-index-to-llvm")
+
+  mod = ApplyRegisteredPass(mod, "convert-func-to-llvm")
+  func = Match(mod, ops={"func.func"})
+  func = ApplyRegisteredPass(func, "convert-arith-to-llvm")
+  func = ApplyRegisteredPass(func, "convert-cf-to-llvm")
+  func = ApplyRegisteredPass(func, "convert-ub-to-llvm")
+  func = ApplyRegisteredPass(func, "canonicalize")
+  transform.ApplyCommonSubexpressionEliminationOp(func)
+  mod = ApplyRegisteredPass(mod, "reconcile-unrealized-casts")
+
+  # Anything useful has been lowered by now.
+  # Cleanup IR by removing any dead symbols.
+  # This step aims to avoid errors caused by frontend leftovers.
+  # See issue: #704
+  transform.ApplyDeadCodeEliminationOp(mod)
+
+  if "llvm" in config["dump"]:
+    transform.PrintOp(target=mod, name="DUMP: stage-llvm")
+
+  return mod
+
+
+def MainSchedule(**config):
+  module = ir.Module.create()
+  module.operation.attributes["transform.with_named_sequence"] = (
+    ir.UnitAttr.get()
+  )
+  with InsertionPoint(module.body):
+    named_sequence = transform.NamedSequenceOp(
+      "__transform_main",
+      [transform.AnyOpType.get()],  # input types
+      [],  # output types
+      # [transform.AnyOpType.get()],  # output types
+      arg_attrs=[{"transform.readonly": ir.UnitAttr.get()}],
+    )
+    with InsertionPoint(named_sequence.body):
+      func = Match(named_sequence.bodyTarget, ops={"func.func"})
+      mod = transform.GetParentOp(
+        transform.AnyOpType.get(),
+        func,
+        op_name="builtin.module",
+        deduplicate=True,
+      )
+
+      try:
```

Comment:
@thiagotei, @fschlimb this (utterly standard) `try ... except ...` allows me to place a `raise PipelineInterrupt` line between any two transform ops above and it appropriately cuts off schedule generation right there. As we caught the exception, the `print(module)` line at the end of `main` still gets executed.

(In more complex setups, e.g. when generating named sequences this is unlikely to be as simple.)